searchlib/src/tests/diskindex/diskindex/diskindex_test.cpp

```diff
@@ -199,45 +199,44 @@ DiskIndexTest::requireThatLookupIsWorking(const EmptySettings& empty_settings)
     uint32_t f1(_schema.getIndexFieldId("f1"));
     uint32_t f2(_schema.getIndexFieldId("f2"));
     uint32_t f3(_schema.getIndexFieldId("f3"));
-    LookupResult::UP r;
-    r = _index->lookup(f1, "not");
-    EXPECT_TRUE(!r || r->counts._numDocs == 0);
+    auto r = _index->lookup(f1, "not");
+    EXPECT_TRUE(r.counts._numDocs == 0);
```

Comment:
Consider using `EXPECT_EQ()` instead. Applies for changes further down as well (as `!r ||` is removed).